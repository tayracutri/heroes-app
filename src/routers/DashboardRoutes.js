import { Routes, Route } from 'react-router-dom';
import { Navbar } from '../components/ui/Navbar';
import { DcScreen } from '../components/dc/DcScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { HeroesScreen } from '../components/heroes/HeroesScreen';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container">
                <Routes>
                    <Route path="marvel" element={<MarvelScreen />} />
                    <Route path="dc" element={<DcScreen />} />

                    <Route path="search" element={<SearchScreen />} />
                    <Route path="hero/:heroeId" element={<HeroesScreen />} />

                    <Route path="/" element={<MarvelScreen />} />

                </Routes>
            </div>
        </>
    )
}