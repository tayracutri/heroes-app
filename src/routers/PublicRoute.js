import React, { useContext } from 'react'
import { Navigate } from 'react-router-dom';
import { AuthContext } from '../components/auth/AuthContext';

export const PublicRoute = ({ children }) => {
    const { user } = useContext(AuthContext)

    console.log(children);

    return user.logged
        ? <Navigate to='login' />
        : children
}


