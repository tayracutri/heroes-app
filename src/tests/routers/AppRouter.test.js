import { AppRouter } from "../../routers/AppRouter"
import { mount } from 'enzyme'
import { AuthContext } from "../../components/auth/AuthContext";



describe('test on AppRouter', () => {
    const value = {
        dispatch: jest.fn(),
        user: { logged: false }
    }

    test('should show the loginscreen if the user isnt auth', () => {


        const wrapper = mount(
            <AuthContext.Provider value={value}>

                <AppRouter />

            </AuthContext.Provider>
        )
        // console.log(wrapper.html());

        expect(wrapper.find('div').exists()).toBe(true);
        expect(wrapper.find('h1').text().trim()).toBe('Login Screen');
        expect(wrapper).toMatchSnapshot();
    });

    test('should show the marvelscreen if the user is auth', () => {

        const value = {
            dispatch: jest.fn(),
            user: { name: 'Tayra', logged: true }
        }
        const wrapper = mount(
            <AuthContext.Provider value={value}>

                <AppRouter />

            </AuthContext.Provider>
        )

        expect(wrapper.find('.navbar').exists()).toBeTruthy();

    });
});

