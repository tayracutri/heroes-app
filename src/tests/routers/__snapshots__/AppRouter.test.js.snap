// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on AppRouter should show the loginscreen if the user isnt auth 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
     Log In 
  </button>
</div>
`;
